company_name_train.py

```diff
@@ -1,48 +1,90 @@
-df_tmp = df_tmp[df_tmp["name_allowed"] == True]
-
-df_tmp["len_name"] = [len(name) for name in df_tmp.name]
-# remove too long names, hard to encode and we are not interested in them
-df_tmp = df_tmp[df_tmp.len_name <= 25]
-
-X = df_tmp.name.to_list()
-X_dummy = ["google"] * 100
-rnn_vae = RnnVae(X, GO="A", EOS="Z", latent_dim=128, intermediate_dim=256)
-rnn_vae_dummy = RnnVae(X_dummy, GO="A", EOS="Z", latent_dim=128, intermediate_dim=256)
-rnn_vae.train(epochs=1, batch_size=16)
-rnn_vae_dummy.train(epochs=30, batch_size=16)
-
-# dummy demo
-word = "google"
-for _ in range(1000):
-    sampled_around = rnn_vae_dummy.sample_around(word)
-    if sampled_around != word:
-        print(sampled_around)
-
-# normal demo
-# problem is now we always sample around the same data point distribution: 0 mean and 1 std.
-for _ in range(30):
-    print(rnn_vae.sample_prediction())
+
+def select_names():
+    df = pd.read_csv("~/datasets/companies_sorted.csv", sep=",")
+    df = df[df.industry.isin(["information technology and services", "internet"])]
+    df = df[df["current employee estimate"] > 5]
+    df.dropna(subset=["name"], inplace=True)
+
+    def allow_name(name):
+        for token in name:
+            try:
+                token.encode("ascii")
+            except:
+                return False
+        return True
+
+    # remove chinese names and others that use characters we do not want to use
+    df["name_allowed"] = [allow_name(name) for name in df.name]
+    df = df[df["name_allowed"] == True]
+
+    df["len_name"] = [len(name) for name in df.name]
+    # remove too long names, hard to encode and we are not interested in them
+    df = df[df.len_name <= 12]
+    return df.name.to_list()
+
+
+def current_date():
+    current_date = datetime.datetime.now()
+    return current_date.strftime("%Y-%m-%d-%H%M")
+
+
+def make_dir(output_path):
+    try:
+        os.makedirs(output_path)
+    except FileExistsError:
+        pass
+    except Exception as e:
+        raise e
+
+
+if __name__ == "__main__":
+    argparser = argparse.ArgumentParser()
+    argparser.add_argument("--epochs", default=1, type=int, help="")
+    argparser.add_argument("--batch-size", default=16, type=int, help="")
+    argparser.add_argument("--latent-dim", default=128, type=int, help="")
+    argparser.add_argument("--intermediate-dim", default=256, type=int, help="")
+    argparser.add_argument(
+        "--save-pred-every",
+        default=5,
+        type=int,
+        help="save model predictions every x epochs",
+    )
+    argparser.add_argument(
+        "--predictions", default=100, type=int, help="number of predictions to make"
+    )
+
+    args = argparser.parse_args()
+    df_preds = pd.DataFrame()
+    current_date = current_date()
+    output_path = os.path.join(os.path.expanduser("~"), "tmp", current_date)
+    make_dir(output_path)
+    names = select_names()
+    rnn_vae = RnnVae(
+        names,
+        GO="A",
+        EOS="Z",
```

Comment:
"A" and "Z" are not in the dataset tokens. Since input dataset is assumed to be unique for this projet now it makes sense to hardcode them. If projet has to work with several datasets in the future could be an argument or determined dynamically